anilist.py

```diff
@@ -172,54 +172,78 @@ def match_to_plex(anilist_series: List[AnilistSeries], plex_series_watched: List
-            if custom_mapping_season_count > 1:
-                logger.warning(
-                    "[ANILIST] Found same custom mapping id for multiple seasons "
-                    "so not using per season processing but updating as one | "
-                    f"title: {plex_title} | anilist id: {custom_mapping_seasons_anilist_id} | "
-                    f"total watched episodes: {plex_watched_episode_count_custom_mapping}"
+
+                    custom_mapped_seasons.append(plex_season.season_number)
+                    match = next(
+                        (item for item in anilist_matches if item['anilist_id'] == matched_id),
+                        None,
+                    )
+
+                    if not match:
+                        anilist_matches.append({
+                            "anilist_id": matched_id,
+                            "watched_episodes": plex_season.watched_episodes,
+                            "total_episodes": plex_season.last_episode,
+                            "mapped_seasons": [plex_season.season_number],
+                            "duplicate_mappings": 0,
+                        })
+                        continue
+                    # For multiple seasons with the same anilist id check how they should be added together
+                    elif (index != 0 and plex_season.first_episode > plex_seasons[index - 1].last_episode):
+                        match["watched_episodes"] = plex_season.watched_episodes
+                    else:
+                        match["watched_episodes"] += plex_season.watched_episodes
+
+                        logger.warning(f"[MAPPING] Duplicate anilist-id found for custom mapping | "
+                                       f"title: {plex_title} | "
+                                       f"anilist id: {matched_id} | "
+                                       f"total watched episodes: {match['watched_episodes']} | "
+                                       f"previous season matches: {match['mapped_seasons']} | "
+                                       f"current season: {plex_season.season_number} |"
+                                       )
+
+                    match["total_episodes"] = + plex_season.last_episode
+                    match["mapped_seasons"].append(plex_season.season_number)
+                    match["duplicate_mappings"] += 1
+
+            for match in anilist_matches:
+                logger.info(
+                    "[MAPPING] Custom Mapping of Title found | "
+                    f"title: {plex_title} | anilist id: {match['anilist_id']} | "
+                    f"total watched episodes: {match['watched_episodes']} | "
+                    f"seasons mapped: {match['mapped_seasons']} | "
                 )
+                # If we had custom mappings for multiple seasons with the same ID use
+                # cumulative episode count and skip per season processing
+                if match["duplicate_mappings"] == 0:
```

Comment:
You're skipping the add_or_update for seasons where 1 season is mapped to 1 anilist id.

Bleach right now has 16 TVDB seasons mapped to anilist id 269 and 1 season to anilist id 116674. The author of the linked issue also has this situation with Fairy Tail season 8.

This loop doesn't update the episodes because of this condition, the loop below doesn't update because the season was already added to `custom_mapped_seasons`.

Reply:
Ah my bad.
I kept this condition because it was there originally but didn't really understand if there was something I was missing.
If this condition is not supposed to exist it actually makes the code a lot simpler.
Can remove the duplicate `parameter` entirely